### ISSUE TITLE
won't build for macOS with error

### ISSUE THREAD

**Issue by copypasteearth** [authpass/macos_secure_bookmarks#1]
Im getting this error when trying to run on macOS, some help would be appreciated

```
SecureBookmarksPlugin.swift:60:15: error: initializer for conditional binding must
have Optional type, not 'URL'
[        ]         guard let url = try URL(resolvingBookmarkData: bookmark, options: .withSecurityScope, bookmarkDataIsStale: &isStale) else {
[        ]               ^

```

**Comment by hpoul** [comment]
have you tried starting the example/ project? which swift version are you using?

**Comment by copypasteearth** [comment]
I just downloaded the repository and ran the example and it worked, there is something messed up in my project I guess. I actually messed it up to the point that none of my plugins are compiling now so I have to wait for my issue in the flutter repository to be resolved before I figure out what the problem was with my project with this plugin. I will come back and update this thread when I figure it out.

**Comment by copypasteearth** [comment]
UPDATE: I just copied the macOS folder from the example project to my projects folder and it built and worked. There is something that is preventing it from being built with the flutter create . folder that is automatically generated by flutter. maybe this plugin was built with an older version of flutter. I got it working but would really like it to work with the latest macOS folder structure

**Comment by copypasteearth** [comment]
UPDATE 2: I was messing around with the plugin code and recreated the example folder so that it has the latest macOS folder and I guess the lates swift, then I put a question mark after the `try?` in the code in my opening post and it built and ran the same as the older version

`guard let url = try? URL(resolvingBookmarkData: bookmark, options: .withSecurityScope, bookmarkDataIsStale: &isStale) else {

**Comment by copypasteearth** [comment]
#2 

submitted a pull request

**Comment by hpoul** [comment]
Thanks for the PR, unfortunately i've already started playing around myself.. i've just pushed a commit into master https://github.com/authpass/macos_secure_bookmarks/commit/1322ce8981b203737334a1ef12617747831a94f7

maybe you could give this one a try as well? then i'll release a new version to pub

**Comment by copypasteearth** [comment]
nice works perfect, I just tested it

**Comment by hpoul** [comment]
thanks again, published as 0.1.1 🎉️
